client/http_sender.go

```diff
@@ -123,22 +120,20 @@ func (s *httpSender) post(call proto.Call) error {
 
 		resp, err := s.client.Do(req)
 		if err != nil {
-			return retry.Continue, err
+			continue
 		}
-
 		defer resp.Body.Close()
```

Comment:
Defers work at function scope, not block scope. With this pattern response bodies won't get closed until the last retry has completed. (and in go's http client, failing to close a response body prevents that connection from being reused). When a retry loop contains a defer, I think we need to refactor the body into a separate function.